Makefile

```diff
@@ -1,125 +1,152 @@
+TEST_HARNESS_REPO := https://github.com/anchore/test-infra.git
 
-# Environment variables set in CircleCI environment
+
+#### CircleCI environment variables
+############################################################
 export VERBOSE ?= false
 export CI ?= false
-# DOCKER_USER & DOCKER_PASS are declared in CircleCI contexts
 export DOCKER_PASS ?=
 export DOCKER_USER ?=
-# LATEST_RELEASE_BRANCH is declared in the CircleCI project environment variables settings
 export LATEST_RELEASE_BRANCH ?=
 export PROD_IMAGE_REPO ?=
 export RELEASE_BRANCHES ?=
 
+# without the following line, the line below
+# (ACTIVATE_VENV := source $(VENV)/bin/activate)
+# fails, oddly. when i change source to . it works
+# even if SHELL := ... is commented out
+#SHELL := /usr/bin/env bash
+
 # Use $CIRCLE_BRANCH if it's set, otherwise use current HEAD branch
 GIT_BRANCH := $(shell echo $${CIRCLE_BRANCH:=$$(git rev-parse --abbrev-ref HEAD)})
+
 # Use $CIRCLE_PROJECT_REPONAME if it's set, otherwise the git project top level dir name
 GIT_REPO := $(shell echo $${CIRCLE_PROJECT_REPONAME:=$$(basename `git rev-parse --show-toplevel`)})
 TEST_IMAGE_NAME := $(GIT_REPO):dev
+
 # Use $CIRCLE_SHA if it's set, otherwise use SHA from HEAD
 COMMIT_SHA := $(shell echo $${CIRCLE_SHA:=$$(git rev-parse HEAD)})
+
 # Use $CIRCLE_TAG if it's set
 GIT_TAG ?= $(shell echo $${CIRCLE_TAG:=null})
 
-# Make environment configuration
+CLUSTER_NAME := e2e-testing
+
+
+# Environment configuration for make
+############################################################
 VENV := venv
+ACTIVATE_VENV := . $(VENV)/bin/activate
 PYTHON := $(VENV)/bin/python3
-.DEFAULT_GOAL := help # Running `Make` will run the help target
-.NOTPARALLEL: # wait for targets to finish
+CI_USER := circleci
+
+# Running make will invoke the help target
+.DEFAULT_GOAL := help
 
-# RUN_TASK is a wrapper script used to invoke commands found in scripts/ci/make/*_tasks
-# These scripts are where all individual tasks for the pipeline belong
-RUN_TASK := scripts/ci/run_make_task
+# Run make serially. Note that recursively invoked make will still
+# run recipes in parallel (unless they also contain .NOTPARALLEL)
+.NOTPARALLEL:
 
+# Specify the Docker image to use in CI commands
+# CI_RUNNER_IMAGE := docker.io/anchore/test-infra:python36
+CI_RUNNER_IMAGE := test-infra:python36
 
-# Define available make commands -- use ## on target names to create 'help' text
+CI_CMD := anchore-ci/local_ci
 
-.PHONY: ci ## run full ci pipeline locally
-ci: VERBOSE := true
-ci: lint build test push
 
-.PHONY: build
-build: Dockerfile ## build dev image
-	@$(RUN_TASK) build "$(COMMIT_SHA)" "$(TEST_IMAGE_NAME)"
+#### Make targets
+############################################################
 
-.PHONY: cluster-up
-cluster-up: venv tests/e2e/kind-config.yaml ## run kind testing k8s cluster
-	@$(RUN_TASK) install_cluster_deps "$(VENV)"
-	@$(RUN_TASK) kind_cluster_up "$(VENV)"
+.PHONY: all venv install install-dev build clean printvars help
```

Comment:
Nit: I prefer putting the .PHONYs above each individual target, it creates more lines in the Makefile but I think it's much easier to see exactly which targets are phony.

Reply:
Let's talk about it. I find that it decreases readability.